spec/acceptance/clock_spec.rb

```diff
@@ -22,6 +22,21 @@
     end
   end
 
+  describe "Runs the callback" do
+    let(:logger) { instance_double("Logger") }
+    around do |example|
+      old_logger = Clockwork::Test.manager.config[:logger]
+      Clockwork::Test.configure { |config| config[:logger] = logger }
+      example.run
+      Clockwork::Test.configure { |config| config[:logger] = old_logger }
+    end
+
+    it "runs the defined callback" do
+      expect(logger).to receive(:info).with("Running before tick")
+      Clockwork::Test.run(file: clock_file, max_ticks: 1)
```

Comment:
This is personal opinion, but to make it more clear to preserve the 4-phase test pattern of setup, exercise, verify, teardown, I'd prefer to use `allow` and `have_received`:

```ruby
allow(logger).to receive(:info)

Clockwork::Test.run(file: clock_file, max_ticks: 1)

expect(logger).to have_received(:info).with("Running before tick").once
```

I know it's more verbose, but I find it easier to reason about.

Adding that `once` should aid in my question above about the callback being registered (and run) on both managers.

Reply:
Make sense. Will change.